app/queries/sortered_products.rb

```diff
@@ -0,0 +1,53 @@
+class SorteredProducts < Rectify::Query
+  include PaginationHelper
+
+  def initialize(params)
+    @params = params
+    @params[:sort] = 'newest' unless @params[:sort].present?
+    @page = @params[:page].present? ? @params[:page].to_i : 1
+    @limit = 12
+    @products = Product.in_stock
+  end
+
+  def query
+    all.limit(@limit).offset(offset).decorate
+  end
+
+  def total_pages
+    total = all.to_a.size
+    (total.to_f / @limit).ceil
+  end
+
+  private
+    def all
+      @products = @products.where(category: @params[:category]) if @params[:category].present?
+      case @params[:sort]
```

Comment:
1. @products = should be moved before case
2. You can create methods that will be called 'newest', 'popular' etc in the orders, than you just need to check if method is allowed and use send_public(method_name)